app/pages/challenges/components/challengeForm.tsx

```diff
@@ -24,9 +34,86 @@ export const ChallengeForm: React.FC<ChallengeFormProps> = ({ userId }) => {
     goal: "",
     unit: "",
   });
-  const [friendIds, setFriendIds] = useState<string>("");
-  const [invitedFriends, setInvitedFriends] = useState<string[]>([]);
+  const [invitedFriends, setInvitedFriends] = useState<FriendOption[]>([]);
   const [errors, setErrors] = useState<Record<string, string>>({});
+  const [isSearchOpen, setIsSearchOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const searchRef = useRef<HTMLDivElement>(null);
+  const [debouncedSearchTerm, setDebouncedSearchTerm] = useState("");
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setDebouncedSearchTerm(searchTerm);
+    }, 300);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [searchTerm]);
```

Comment:
close close . please dont forget the code you have written :) 

<img width="923" alt="Screenshot 2025-03-04 at 21 33 39" src="https://github.com/user-attachments/assets/ee40d7ca-a279-4a28-a982-1000412a5bbe" />


use debounce from lodash instead of cleartimeout